app/src/main/java/io/gnosis/safe/ui/transactions/details/ButtonStateHelper.kt

```diff
@@ -0,0 +1,66 @@
+package io.gnosis.safe.ui.transactions.details
+
+class ButtonStateHelper(
+    val hasBeenRejected: Boolean,
+    val needsYourConfirmation: Boolean,
+    val isRejection: Boolean,
+    val needsExecution: Boolean,
+    val canReject: Boolean,
+    val isOwner: Boolean,
+    val completed: Boolean
+) {
+
+    fun rejectButtonIsVisible(): Boolean {
+        if (!isOwner || completed) {
```

Comment:
not critical: it would be compacter with when {...

Reply:
Done